src/main.js

```diff
@@ -107,17 +130,41 @@ export class Draggable {
         this._mouseup = (e) => {
             unbind(this.document, "mousemove", this._mousemove);
             unbind(this.document, "mouseup", this._mouseup);
+
+            if (this.clickMoveClick){
+                unbind(this.document, "keydown", this._keydown);
+                unbind(this.document, "contextmenu", preventDefault);
+            }
+
             this._releaseHandler(e);
         };
 
         this._pointerdown = (e) => {
             if (e.isPrimary && e.button === 0) {
-                bind(this.document, "pointermove", this._pointermove);
-                bind(this.document, "pointerup", this._pointerup);
-                bind(this.document, "pointercancel", this._pointerup);
-                bind(this.document, "contextmenu", preventDefault);
 
-                this._pressHandler(e);
+                if (!this._clickMoveClick) {
+                    bind(this.document, "pointerup", this._pointerup);
+                    bind(this.document, "pointercancel", this._pointerup);
+                    bind(this.document, "contextmenu", preventDefault);
+                    bind(this.document, "pointermove", this._pointermove);
+                    this._pressHandler(e);
+                } else {
+                    if (this._isDragging) {
+                        bind(this.document, "pointerup", this._pointerup);
+                        bind(this.document, "pointercancel", this._pointerup);
+                    } else {
+                        bind(this.document, "pointermove", this._pointermove);
+                        bind(this.document, "keydown", this._keydown);
+                        this._pressHandler(e);
+                    }
+
+                    this._isDragging = !this._isDragging;
+                }
+            }
+
+            if (this._clickMoveClick && e.button == 2) {
```

Comment:
We can extract an enum like `MouseButton.Left | Right | Middle` as numbers like `2` are not very semantic.